Enterprise-CMCS/macpro-mdct-mfp
# 2984  [Code Refactor]

### Description
[ This is a code refactor of the 2984 ticket ] 
Goal: New "Other" Target Populations should be displayed on the 'Define Initiative'  but the field data shouldn't be mutated. 

The 'Define Initiative' page now reflects “other target populations” that are entered in the 'Benchmark Projections' section. There's only target population logic ( repeatableTargetPopulationChoices function in form.ts) to fufill the ticket but the *getRepeatableChoiceLists* function can be used as a utility to create other repeat other choice lists.

<img width="751" alt="Screenshot 2023-10-04 at 2 07 47 PM" src="https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/14514294/62afdbf1-93eb-4c50-b6e1-5f57f6a559d7">

<img width="607" alt="Screenshot 2023-10-04 at 2 08 08 PM" src="https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/14514294/8c2da4ba-f4b4-462b-bc9c-cf9c357ba7e6">

https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/14514294/1624fa75-5902-4056-a5fb-d5c4e94afbca


### Related ticket(s)
MDCT-[https://qmacbis.atlassian.net/browse/MDCT-2813]

---
### How to test
1. Go to Transition Benchmark Projections
2. Create new target population and save
3. Go to State and Territory-Specific Initiatives: I. Define initiative
4. In the Target Populations section there should be a new Other option


---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
